tests/integration/test_schema.py

```diff
@@ -2356,11 +2356,13 @@ async def test_schema_hard_delete_whole_schema(registry_async_client: Client) ->
 
 async def test_schema_hard_delete_and_recreate(registry_async_client: Client) -> None:
     subject = create_subject_name_factory("test_schema_hard_delete_and_recreate")()
+    schema_name = create_schema_name_factory("test_schema_hard_delete_and_recreate")()
+
     res = await registry_async_client.put("config", json={"compatibility": "BACKWARD"})
     assert res.status == 200
     schema = {
         "type": "record",
-        "name": "myenumtest",
+        "name": schema_name,
```

Comment:
Why is this necessary?

Reply:
I'm getting a failure here with SR:

```
sudo docker-compose -f ./tests/integration/confluent-docker-compose.yml up -d
pytest tests/integration/test_schema.py::test_schema_hard_delete_and_recreate --registry-url http://localhost:8081 --rest-url http://localhost:8082/ --kafka-bootstrap-servers localhost:9092
```

Results in:
```
tests/integration/test_schema.py:2415: in test_schema_hard_delete_and_recreate
    assert schema_id == res.json()["id"], "after permanent deleted the same schema registered, the same identifier"
E   AssertionError: after permanent deleted the same schema registered, the same identifier                                                                                                                                                   
E   assert 1 == 2 
```

Reply:
> Why is this necessary?

The state in SR that is persisted across test runs messes things up. It's cleaner to always have a fresh schema name (and no such schema in SR at the beginning of the test).

Try this. 

Do as you did, but change first `myenumtest` to something else, let's say `s1`.  Run it. Fails. Run again. Succeeds!

Reply:
It's supposed to fail against SR, I mentioned that in the PR. That's a small incompatibility between SR and Karapace and should probably be fixed. I haven't checked how to fix it.

Reply:
> It's supposed to fail against SR, I mentioned that in the PR. That's a small incompatibility between SR and Karapace and should probably be fixed. I haven't checked how to fix it.

My bad, I went mechanically through every PR and didn't pay attention to what you wrote. Thanks.

Reply:
> I haven't checked how to fix it.

On a hard delete we should delete the schema from the `schemas` dictionary. Then the schema should not be found when sent again:

https://github.com/aiven/karapace/blob/9dd407d69cfb5cf61b48df394b8d55ce0d50ac86/karapace/schema_reader.py#L196-L204

It would be interesting to see what happens when we do this:

- Start SR
- Publish schema / soft delete / hard delete
- Restart SR
- Publish the same schema